### PR TITLE
XS✔ ◾ split migration rule into two

### PR DESCRIPTION
**Pull request by lukecookssw**
This PR creates two rules which will supersede the "Do you know how to handle complex .NET migrations?" rule:

- Do you create a migration plan?
- Migrating Web Apps to .NET Core

I **have not** added these rules to a category yet, as we are waiting on #6580 to be merged so we have the appropriate category to link these rules under. I'm happy for these rules to be "orphaned" for now, but let me know if that's a problem.

### CODE REVIEW

File: rules/do-you-create-a-migration-plan/rule.md [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+  - title: Yazhi Chen
+    url: https://www.ssw.com.au/people/yazhi-chen
+related:
+  - dotnet-upgrade-assistant
+  - migrate-from-system-web-to-modern-alternatives
+created: 2023-09-06T23:08:53.979Z
+guid: d47bb1e4-261f-436e-84fc-fdb1b21e0d36
+---
+
+Migrating from .NET Framework to .NET Core brings huge advantages to your app's performance, hosting fees, and maintainability. But it's important that you understand what the road to .NET Core looks like for your app *before* you start breaking things! So how do you ensure your migration is being done the right way?
```
Comment (wicksipedia) [inline rules/do-you-create-a-migration-plan/rule.md:19]:
```suggestion
Migrating from .NET Framework (4.x) to the latest .NET (5+) brings huge advantages to your app's performance, hosting fees, and maintainability. But it's important that you understand what the road to .NET 5+ looks like for your app *before* you start breaking things! So how do you ensure your migration is being done the right way? 
```

File: rules/do-you-create-a-migration-plan/rule.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+created: 2023-09-06T23:08:53.979Z
+guid: d47bb1e4-261f-436e-84fc-fdb1b21e0d36
+---
+
+Migrating from .NET Framework to .NET Core brings huge advantages to your app's performance, hosting fees, and maintainability. But it's important that you understand what the road to .NET Core looks like for your app *before* you start breaking things! So how do you ensure your migration is being done the right way? 
+
+<!--endintro-->
+
+# Preparation
+The migration from .NET Framework to .NET Core is the perfect opportunity for you to take stock of your current application architecture, and address any technical debt your app has accumulated. Trying to migrate an application that's poorly architected or carrying a lot of tech debt will make your migration **exponentially harder**. Therefore, now is the time to perform a full audit of your app and ensure you have PBIs to rectify these problems before you consider it "migration-ready".
```
Comment (wicksipedia) [inline rules/do-you-create-a-migration-plan/rule.md:24]:
```suggestion
The migration to a newer versions of .NET is the perfect opportunity for you to take stock of your current application architecture, and address any technical debt your app has accumulated. Trying to migrate an application that's poorly architected or carrying a lot of tech debt will make your migration **exponentially harder**. Therefore, now is the time to perform a full audit of your app and ensure you have PBIs to rectify these problems before you consider it "migration-ready".
```

File: rules/do-you-create-a-migration-plan/rule.md [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+
+### Dependency analysis
+Imagine a typical N-tiered application. Over the course of that application's life, the lines between each tier will often get blurred (either accidentally or deliberately). This can result in certain dependencies appearing in places they shouldn't - such as `System.Web` references showing up in your application or data layer. This is a very common code smell and is a great starting point to cleaning up your app.
+
+If your app has 3rd party dependencies (e.g. with a financial system, reporting system, etc.) - now is the time to do some investigation around those integration points to determine whether those services provide .NET Core libraries and how those libraries differ (if at all). Create PBIs for these as well.
```
Comment (wicksipedia) [inline rules/do-you-create-a-migration-plan/rule.md:29]:
```suggestion
If your app has 3rd party dependencies (e.g. with a financial system, reporting system, etc.) - now is the time to do some investigation around those integration points to determine whether those services provide compatible libraries and how those libraries differ (if at all). Create PBIs for these as well. 
```

File: rules/do-you-create-a-migration-plan/rule.md [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+
+### Dependency analysis
+Imagine a typical N-tiered application. Over the course of that application's life, the lines between each tier will often get blurred (either accidentally or deliberately). This can result in certain dependencies appearing in places they shouldn't - such as `System.Web` references showing up in your application or data layer. This is a very common code smell and is a great starting point to cleaning up your app.
+
+If your app has 3rd party dependencies (e.g. with a financial system, reporting system, etc.) - now is the time to do some investigation around those integration points to determine whether those services provide .NET Core libraries and how those libraries differ (if at all). Create PBIs for these as well. 
+
+### Infrastructure
+If you host your app on premise, it's also worth checking your infrastructure to ensure it's got the necessary .NET Core runtimes.
```
Comment (wicksipedia) [inline rules/do-you-create-a-migration-plan/rule.md:32]:
```suggestion
If you host your app on premise, it's also worth checking your infrastructure to ensure it's has the necessary runtimes.
```

File: rules/do-you-create-a-migration-plan/rule.md [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+
+and your other projects using
+
+```bash
+try-convert --keep-current-tfms
+```
+:::
+
+
+### Change all your projects to be able to target multiple Target framework monikers (TFM)
```
Comment (wicksipedia) [inline rules/do-you-create-a-migration-plan/rule.md:67]:
```suggestion
### Change all your projects to be able to target multiple Target Framework Monikers (TFM)
```

File: rules/do-you-create-a-migration-plan/rule.md [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+Now you have shiny new SDK-style `csproj` files, it's time to see what breaks!
+
+Targeting both your current .NET Framework version *and* your future .NET Core version will give you the following information:
```
Comment (wicksipedia) [inline rules/do-you-create-a-migration-plan/rule.md:70]:
```suggestion
Targeting both your current .NET Framework version *and* your future .NET version will give you the following information:
```

File: rules/do-you-create-a-migration-plan/rule.md [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+
+* Expose any build errors you receive when trying to build for .NET Core
```
Comment (wicksipedia) [inline rules/do-you-create-a-migration-plan/rule.md:72]:
```suggestion
* Expose any build errors you receive when trying to build for .NET
```

File: rules/do-you-create-a-migration-plan/rule.md [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+* Expose any build errors you receive when trying to build for .NET Framework
+
+::: greybox
+Why is this important?
+
+Imagine you *don't* do this, and instead, you simply target .NET Core. You get a list of 100 build errors due to breaking changes - too many for 1 Sprint (or 2 Sprints, or 3).
```
Comment (wicksipedia) [inline rules/do-you-create-a-migration-plan/rule.md:78]:
```suggestion
Imagine you *don't* do this, and instead, you simply target there newer version of .NET. You get a list of 100 build errors due to breaking changes - too many for 1 Sprint (or 2 Sprints, or 3).
```

File: rules/do-you-create-a-migration-plan/rule.md [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+
+You start fixing these build errors. You go from 100 errors to 50 - progress!
+Then you're told there's an urgent bug/feature/whatever that needs fixing ASAP. But you've still got 50 build errors when you're targeting .NET Core.
```
Comment (wicksipedia) [inline rules/do-you-create-a-migration-plan/rule.md:81]:
```suggestion
Then you're told there's an urgent bug/feature/whatever that needs fixing ASAP. But you've still got 50 build errors when you're targeting .NET.
```

File: rules/do-you-create-a-migration-plan/rule.md [hunk 10]
```diff
@@ -0,0 +1,226 @@
...
+
+"No problem", you say. "I'll just switch back to .NET Framework and do this fix, and push out a new deployment".
+
+You switch to .NET Framework, build the project, and...25 build errors?!
+
+While you were fixing those .NET Core build errors, you wrote code that isn't compatible with .NET Framework. Now you have an urgent bug/feature/whatever, as well as 25 new build errors you have to solve ☠️.
```
Comment (wicksipedia) [inline rules/do-you-create-a-migration-plan/rule.md:87]:
```suggestion
While you were fixing those build errors, you wrote code that isn't compatible with .NET Framework. Now you have an urgent bug/feature/whatever, as well as 25 new build errors you have to solve ☠️.
```

File: rules/do-you-create-a-migration-plan/rule.md [hunk 11]
```diff
@@ -0,0 +1,226 @@
...
+
+Using multiple TFMs from day 1 ensures you are fixing the breaking changes for .NET Core, *without* introducing breaking changes in .NET Framework.
```
Comment (wicksipedia) [inline rules/do-you-create-a-migration-plan/rule.md:89]:
```suggestion
Using multiple TFMs from day 1 ensures you are fixing the breaking changes for .NET, *without* introducing breaking changes in .NET Framework.
```

**Review by wicksipedia — COMMENTED** [review]
.NET Core ended with 3.1

Since v5 - it is all just .NET now (and I got tired of fixing the Core references)

File: rules/do-you-create-a-migration-plan/rule.md [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+
+### Dependency analysis
+Imagine a typical N-tiered application. Over the course of that application's life, the lines between each tier will often get blurred (either accidentally or deliberately). This can result in certain dependencies appearing in places they shouldn't - such as `System.Web` references showing up in your application or data layer. This is a very common code smell and is a great starting point to cleaning up your app.
+
+If your app has 3rd party dependencies (e.g. with a financial system, reporting system, etc.) - now is the time to do some investigation around those integration points to determine whether those services provide compatible libraries and how those libraries differ (if at all). Create PBIs for these as well. 
+
+### Infrastructure
+If you host your app on premise, it's also worth checking your infrastructure to ensure it's has the necessary runtimes.
```
Comment (GordonBeeming) [inline rules/do-you-create-a-migration-plan/rule.md:32]:
```suggestion
If you host your app on premise, it's also worth checking your infrastructure to ensure it has the necessary runtimes.
```

File: rules/do-you-create-a-migration-plan/rule.md [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+
+Targeting both your current .NET Framework version *and* your future .NET version will give you the following information:
+
+* Expose any build errors you receive when trying to build for .NET
+* Expose any build errors you receive when trying to build for .NET Framework
+
+::: greybox
+Why is this important?
+
+Imagine you *don't* do this, and instead, you simply target there newer version of .NET. You get a list of 100 build errors due to breaking changes - too many for 1 Sprint (or 2 Sprints, or 3).
```
Comment (GordonBeeming) [inline rules/do-you-create-a-migration-plan/rule.md:78]:
```suggestion
Imagine you *don't* do this, and instead, you simply target the newer version of .NET. You get a list of 100 build errors due to breaking changes - too many for 1 Sprint (or 2 Sprints, or 3).
```

File: rules/do-you-create-a-migration-plan/rule.md [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+
+### Dependency analysis
+Imagine a typical N-tiered application. Over the course of that application's life, the lines between each tier will often get blurred (either accidentally or deliberately). This can result in certain dependencies appearing in places they shouldn't - such as `System.Web` references showing up in your application or data layer. This is a very common code smell and is a great starting point to cleaning up your app.
+
+If your app has 3rd party dependencies (e.g. with a financial system, reporting system, etc.) - now is the time to do some investigation around those integration points to determine whether those services provide compatible libraries and how those libraries differ (if at all). Create PBIs for these as well.
```
Comment (YazhiChen) [inline rules/do-you-create-a-migration-plan/rule.md:29]:
```suggestion
If your app has 3rd party dependencies (e.g. with a financial system, reporting system, etc.) - now is the time to investigate those integration points to determine whether those services provide compatible libraries and how those libraries differ (if at all). Create PBIs for these as well. 
```

**Review by GordonBeeming — APPROVED** [review]
LGTM 🚢

File: rules/do-you-create-a-migration-plan/rule.md [hunk 15]
```diff
@@ -0,0 +1,126 @@
...
+   2. Repeat these steps once the PBIs have been completed related to this project
+
+By the end of this process, you'll have a much clearer view (and backlog!) of your path to the latest .NET:
+
+* PBIs for technical debt
+* PBIs for architectural concerns
+* PBIs for breaking changes
+
+# What's next?
+While this guide aims to give you a high-level view of migrating your app, there are other some special considerations when dealing with complex applications and web apps. Check out these othe rules:
```
Comment (YazhiChen) [inline rules/do-you-create-a-migration-plan/rule.md:122]:
```suggestion
While this guide aims to give you a high-level view of migrating your app, there are other some special considerations when dealing with complex applications and web apps. Check out these other rules:
```

File: rules/do-you-create-a-migration-plan/rule.md [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+uri: migration-plans
+authors:
+  - title: Luke Cook
+    url: https://ssw.com.au/people/luke-cook
+  - title: Gordon Beeming
+    url: https://www.ssw.com.au/people/gordon-beeming
+  - title: Yazhi Chen
+    url: https://www.ssw.com.au/people/yazhi-chen
+related:
+  - dotnet-upgrade-assistant
```
Comment (YazhiChen) [inline rules/do-you-create-a-migration-plan/rule.md:13]: Why are only these two rules listed as related rules here

File: rules/migrating-web-apps/rule.md [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+
+### Create PBIs to identify the upcoming tasks
+
+When a web project is heavily reliant on .NET Framework dependencies, the first step in gauging the effort required for a complete migration is to thoroughly examine these dependencies. This involves a detailed investigation, followed by the creation of PBIs for each dependency. These PBIs serve to accurately scope out the total effort that will be needed for the migration process.
+
+Listed below are rules crafted to aid in the project migration process. Please ensure to incorporate only those rules that are applicable to your specific project.
+
+- [Do you know how to migrate Global.asax to ASP.NET Core?](https://www.ssw.com.au/rules/know-how-to-migrate-global-asax-to-asp-net-core/)
+- [Do you know how to migrate OWIN to ASP.NET Core?](https://www.ssw.com.au/rules/know-how-to-migrate-owin-to-asp-net-core/)
+- [Do you know how to migrate Web.config to ASP.NET Core?](https://www.ssw.com.au/rules/know-how-to-migrate-owin-to-asp-net-core/)
```
Comment (YazhiChen) [inline rules/migrating-web-apps/rule.md:130]: This is probably the previous mistake.
```suggestion
- [Do you know how to migrate Web.config to ASP.NET Core?](https://ssw.com.au/rules/know-how-to-migrate-web-config-to-asp-net-core/)
```

**Review by YazhiChen — CHANGES_REQUESTED** [review]
Please update the typo and update the link for the rule - Do you know how to migrate Web.config to ASP.NET Core?

**Pull request merged**